src/utils/deepCopy.js

```diff
@@ -13,12 +14,7 @@ export const deepCopy = (source) => {
   const keys = Object.keys(source)
 
   keys.forEach((key) => {
-    if (isAccessor(source, key)) {
-      const accessor = Object.getOwnPropertyDescriptor(source, key)
-      Object.defineProperty(newObj, key, accessor)
-    } else {
-      newObj[key] = isObject(source[key]) ? deepCopy(source[key]) : source[key]
-    }
+    newObj[key] = isObject(source[key]) ? deepCopy(source[key]) : source[key]
```

Comment:
```suggestion
    newObj[key] = isObject(source[key]) || isArray(source[key]) ? deepCopy(source[key]) : source[key]
```

It looks to me that if a value is an array we would return the reference. We should probably create a copy here.

I think another approach to this function would to have a base state of an atomic type (not object or array), and then recursively handle objects and arrays.